src/protocol/prf_sharding/mod.rs

```diff
@@ -0,0 +1,704 @@
+use std::iter::repeat;
+
+use futures_util::future::try_join_all;
+
+use super::step::BitOpStep;
+use crate::{
+    error::Error,
+    ff::{Field, GaloisField, Gf2},
+    protocol::{
+        basics::{SecureMul, ShareKnownValue},
+        context::{Context, UpgradableContext, UpgradedContext, Validator},
+        BasicProtocols, RecordId,
+    },
+    repeat64str,
+    secret_sharing::{
+        replicated::{semi_honest::AdditiveShare as Replicated, ReplicatedSecretSharing},
+        BitDecomposed, Linear as LinearSecretSharing,
+    },
+};
+
+pub struct PrfShardedIpaInputRow<BK: GaloisField, TV: GaloisField> {
+    prf_of_match_key: u64,
```

Comment:
Possibly we don't want to replicate the match key PRF with each row record -- we may not need to carry it here at all, as long as we have something like a "match key same as last row" bit.

Reply:
Makes sense to me. I think we can return to this when we implement the first part of the OPRF approach. Since we will be generating the inputs to this stage ourselves - we are free to change this to anything we want.